flink-runtime/src/main/java/org/apache/flink/runtime/blob/TransientBlobCache.java

```diff
@@ -222,7 +221,13 @@ public File getStorageLocation(@Nullable JobID jobId, BlobKey key) throws IOExce
 	}
 
 	private BlobClient createClient() throws IOException {
-		return new BlobClient(serverAddress, blobClientConfig);
+		final InetSocketAddress currentServerAddress = serverAddress;
+
+		if (currentServerAddress != null) {
+			return new BlobClient(serverAddress, blobClientConfig);
```

Comment:
I think you should use `currentServerAddress`, or the creation can fail with an NPE.

Reply:
Definitely. Good catch!